lightning/src/util/config.rs

```diff
@@ -128,6 +135,24 @@ pub struct ChannelHandshakeLimits {
 	///
 	/// Default value: 144, or roughly one day and only applies to outbound channels.
 	pub max_minimum_depth: u32,
+	/// Whether we implicitly trust our own funding transactions to not be double-spent.
+	///
+	/// If this is set, we assume that our own funding transactions are *never* double-spent, and
+	/// thus we can trust them without any confirmations. This is generally a reasonable
+	/// assumption, given we're the only ones who could ever double-spend it (assuming we have sole
+	/// control of the signing keys).
```

Comment:
Should we make the comment even more conservative and say "The funding transaction should not have inputs spendable by a non-trusted third-party" ? The notion of "sole control of the signing keys" can be quickly confusing as there could be an alternative script branch spendable. The "sole knowledge of any satisfying witness of the inputs" would be better imo.

Reply:
Hmm, it already is very explicit - "given we're the only ones who could ever double-spend it". The comment about control of keys is just a parenthetical to make it concrete. I'm not sure its worth clarifying a clarification example.

Reply:
You might have an output locked by a Taproot with 2 script path spends : you own key in the script path but not the ones in the second script path, or they're also locked with a timelock and you think it's safe but the timelock expires. I find the "sole knowledge of any satisfying witness of the inputs" a stronger requirement.